NAS-135360 / 25.10 / Use new editable fields in user form

denysbutenko opened the pull request:
**Changes:**

- Use new editable fields in user form

**Testing:**

Check user form on `Users (WIP)`

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-135360

AlexKarpov98 reviewed (commented):
Form looks cool.
I noticed a couple of issues:

1. Form counts invalid (submit is disabled) until I pass through all the `Additional Details` block.

https://github.com/user-attachments/assets/73e986e8-c935-4a92-84fa-1309ae74d74d


2. Selecting `Shell` leads to the infinite loading input.

https://github.com/user-attachments/assets/12d60d94-de06-4146-88a4-33c6227e20d5

3. I can open edit mode but can't exit it if I don't want to change any value, is that OK?

denysbutenko commented:
> Form counts invalid (submit is disabled) until I pass through all the Additional Details block.

Added asterisk to new labels when field is required.

> Selecting Shell leads to the infinite loading input.

It should not be the case now. 

> I can open edit mode but can't exit it if I don't want to change any value, is that OK? (or if I changed the value I need to open another field for editing to close the previous)

This is a component-specific issue, I will open a ticket for that.

AlexKarpov98 reviewed (commented):
1. That issue still exists, new ticket will be created.
<img width="806" alt="Screenshot 2025-05-02 at 16 00 45" src="https://github.com/user-attachments/assets/f86a5847-ba2c-46cb-91a3-9482df2e528c" />

2. I tried to fill all form items with `*` but the form is still invalid. Seems like if I update `UID` field, it becomes valid.


https://github.com/user-attachments/assets/bbcf2781-cba4-45ba-9c3f-257744ba89d3

denysbutenko commented:
> 1. That issue still exists, new ticket will be created.
> 

I created a [ticket](https://ixsystems.atlassian.net/browse/NAS-135734) for that.

> <img alt="Screenshot 2025-05-02 at 16 00 45" width="806" src="https://private-user-images.githubusercontent.com/22980553/439922949-f86a5847-ba2c-46cb-91a3-9482df2e528c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2MzU0NDIsIm5iZiI6MTc0NjYzNTE0MiwicGF0aCI6Ii8yMjk4MDU1My80Mzk5MjI5NDktZjg2YTU4NDctYmEyYy00NmNiLTkxYTMtOTQ4MmRmMmU1MjhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA3VDE2MjU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmODA5YzBiOGMzYWRlMzQ3YzBhNzNkMWNhNDg2YTJmM2JhNDNmZGQzNzY2MTJmNzE5NDQ4Yzg4ZGI2ZGY2NzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TDGVtMWAKPK_9Di3x7eaqhGedXRVN87xlyyfAgaIf2o">
> 2. I tried to fill all form items with `*` but the form is still invalid. Seems like if I update `UID` field, it becomes valid.
> 
>  Screen.Recording.2025-05-02.at.16.01.22.mov

I updated form to make those fields non-required. I talked with Evgeny, we won't use editable for required fields.

AlexKarpov98 reviewed (approved):
👌 
With it all discussed, we can merge it now.

bugclerk commented:
This PR has been merged and conversations have been locked.
If you would like to discuss more about this issue please use our forums or raise a Jira ticket.
